spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/ReadConf.scala

```diff
@@ -49,12 +53,20 @@ object ReadConf {
     description = """Sets whether to record connector specific metrics on write"""
   )
 
+  val CassandraJoinConcurrentReads = ConfigParameter[Int](
+    name = "spark.cassandra.input.join.concurrent.reads",
+    section = ReferenceSection,
+    default = 5,
```

Comment:
Gut instinct here, but I think this is low I would try benchmarking some higher defaults as well.